js/subtitles-octopus.js

```diff
@@ -361,16 +373,17 @@ var SubtitlesOctopus = function (options) {
                 self.canvas.style.position = 'absolute';
                 self.canvas.style.width = videoSize.width + 'px';
                 self.canvas.style.height = videoSize.height + 'px';
-                self.canvas.style.left = videoSize.x + 'px';
-                var offset = self.canvasParent.getBoundingClientRect().top - self.video.getBoundingClientRect().top;
-                self.canvas.style.top = (videoSize.y - offset) + 'px';
+                self.canvas.style.top = top + 'px';
+                self.canvas.style.left = left + 'px';
                 self.canvas.style.pointerEvents = 'none';
             }
 
             self.worker.postMessage({
                 target: 'canvas',
                 width: self.canvas.width,
-                height: self.canvas.height
+                height: self.canvas.height,
+                top: top,

```

Comment:
Is this sufficient for updating the canvas in the case where there is no video element provided?

Reply:
No need to update top & left inside worker. Canvas inside worker is virtual.